### ISSUE TITLE
[Python] Command line execution of PyArrow and submodules

### ISSUE THREAD

**Issue by pitrou** [apache/arrow#43361]
### Describe the enhancement requested

Currently, attempting to execute PyArrow and its submodules either fails or (misleadingly) does nothing.
```console
$ python -m pyarrow
/home/antoine/t/venv-3.10/bin/python: No module named pyarrow.__main__; 'pyarrow' is a package and cannot be directly executed
$ python -m pyarrow.csv
$ python -m pyarrow.json
$ python -m pyarrow.parquet
/home/antoine/t/venv-3.10/bin/python: No module named pyarrow.parquet.__main__; 'pyarrow.parquet' is a package and cannot be directly executed
```

Some of these at least could do something useful.

### Component(s)

Python

**Comment by pitrou** [comment]
@jorisvandenbossche @danepitkin

**Comment by jorisvandenbossche** [comment]
> Some of these at least could do something useful.

You are imagining eg `python -m pyarrow.parquet <file>` to have some option to print the content of the file or the schema (and same for the other formats) ?

**Comment by pitrou** [comment]
Right. Or perhaps a more sophisticated API if we are so inclined :-)

**Comment by thisisnic** [comment]
This issue has been closed because there has been no activity for the 14 days since the stale warning. Feel free to reopen if this has been closed in error.
